utils/logger/logger.go

```diff
@@ -0,0 +1,205 @@
+// this is a new logger interface for mattermost
+
+package logger
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"runtime"
+
+	l4g "github.com/alecthomas/log4go"
+
+	"github.com/mattermost/platform/model"
+	"github.com/mattermost/platform/utils"
+)
+
+// this pattern allows us to "mock" the underlying l4g code when unit testing
+var debug = l4g.Debug
+var info = l4g.Info
+var err = l4g.Error
+
+func init() {
+	initL4g(utils.Cfg)
+}
```

Comment:
I'm not crazy about lines 17-24. They rely on global state and assume that both Cfg and l4g have already been initialized, which may not be true, depending on when we first try to log a message during startup. Any ideas on how to improve this?